### DISCUSSION TITLE
Get a balance and a list of tokens and Cryptos, How could I do this? For HTML and a Movil App

### DISCUSSION THREAD

**Discussion by Androser** [Web3Auth/Web3Auth#611] [Help]
Hello! I have a really great question! I would like to check if a user has NFT's in their wallet or if the user has some specific cryptocurrencies.
Now the balance only shows the amount of ETH in your account, but it is sure to be able to! We work at HTML, and in Unity we make a version for the iOS and Android cell phone. Can you help us? 

This is the code for Get ETH in HTML. We dont have the code to do something in Unity
```
  const getBalance = async (provider) => {
    const web3 = new Web3(provider);

    // Get user's Ethereum public address
    const address = (await web3.eth.getAccounts())[0];

    // Get user's balance in ether
    const balance = web3.utils.fromWei(
      await web3.eth.getBalance(address) // Balance is in wei
    );

    return balance;
  };
```
[https://d318hu5thvz0lz.cloudfront.net/](url)

**Comment by rinzler99** [comment]
Hey, You can use covalent api to retrieve the custom tokens, custom nfts. And also opensea api to retrieve nft held by specific account address.
https://docs.opensea.io/reference/api-overview
https://www.covalenthq.com/docs/api/#/0/0/USD/1

**Reply by Androser** [reply]
Okay okay, excuse me. I cant understand how it works... And how use it or recibe an answer from HTML

**Reply by rinzler99** [reply]
Exampl code:

```
const options = {
  method: 'GET',
  headers: {accept: 'application/json', 'X-API-KEY': 'your api key'}
};

fetch('https://api.opensea.io/api/v1/assets?order_direction=desc&limit=20&include_orders=false', options)
  .then(response => response.json())
  .then(response => console.log(response))
  .catch(err => console.error(err));
```

**Reply by Androser** [reply]
Please Excuse me, Im really new here, Where can i get a X-API-KEY or my api key?

**Reply by Androser** [reply]
Please Excuse me, Im really new here, Where can i get a X-API-KEY or my api key?

**Comment by Androser** [comment]
Okay okay Ihave ask the XapiKey to OpenSea. They will contact me in max 2 days.  But even I dont have the api Key or HEADER, and I dont know what is that

**Reply by rinzler99** [reply]
To get the information about tokens, nfts owned by an account you will be using apis which need to be authenticated using api keys to access information. To understand more about api, headers, api keys please explore some YouTube videos.

**Reply by Androser** [reply]
Hello! there is any option to do it from Web3Auth?? Here ypu coul get a balance of ETH and could be genial if I can do the same with only other token

**Comment by rinzler99** [comment]
Yes! You can use torus wallet plugin along with web3auth sdk to fetch the balance of custom tokens held by address.

https://web3auth.io/docs/whitelabel/torus-wallet-plugin
